cli/main.rs

```diff
@@ -99,7 +98,62 @@ fn js_check(r: Result<(), ErrBox>) {
   }
 }
 
-// TODO: we might want to rethink how this method works
+fn create_worker_and_state(
+  flags: DenoFlags,
+  argv: Vec<String>,
+) -> (Worker, ThreadSafeState) {
+  use crate::shell::Shell;
+  use std::sync::Arc;
+  use std::sync::Mutex;
+  let shell = Arc::new(Mutex::new(Shell::new()));
+  let progress = Progress::new();
+  progress.set_callback(move |_done, _completed, _total, status, msg| {
+    if !status.is_empty() {
+      let mut s = shell.lock().unwrap();
+      s.status(status, msg).expect("shell problem");
+    }
+  });
+  // TODO(kevinkassimo): maybe make include_deno_namespace also configurable?
+  let state =
+    ThreadSafeState::new(flags, argv, ops::op_selector_std, progress, true)
+      .unwrap();
+  let worker = Worker::new(
+    "main".to_string(),
+    startup_data::deno_isolate_init(),
+    state.clone(),
+  );
+
+  (worker, state)
+}
+
+fn types_command() {
+  let content = include_str!(concat!(
+    env!("GN_OUT_DIR"),
+    "/gen/cli/lib/lib.deno_runtime.d.ts"
+  ));
+  println!("{}", content);
+}
+
+fn print_cache_info(worker: Worker) {
+  let state = worker.state;
+
+  println!(
+    "{} {:?}",
+    ansi::bold("DENO_DIR location:".to_string()),
+    state.dir.root
+  );
+  println!(
+    "{} {:?}",
+    ansi::bold("remote modules cache:".to_string()),
```

Comment:
Nit: the other lines start with capital letters - can you make “remote” capital too?